### PR TITLE
added virtualization provider

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR provides information of the currently used virtualization provider for each installation config. We need this to know which kind of validation and internal business logic to chose from within our G8S microservices.

### CODE REVIEW

File: configuration/provider.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package configuration
+
+type provider string
+
+const (
+	ProviderAWS provider = "aws"
```
Comment (JosephSalisbury) [inline configuration/provider.go]: I would like docstring here

File: configuration/provider.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	ProviderKVM provider = "kvm"
```
Comment (JosephSalisbury) [inline configuration/provider.go]: I would like docstring here

File: configuration/provider.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package configuration
```
Comment (JosephSalisbury) [inline configuration/provider.go]: Need package docstring

Comment (xh3b4sd) [inline configuration/provider.go]: No it doesn't. The package docs are in `configuration.go`. This is not a new package.

File: configuration/provider.go (hunk 1 above)
Comment (xh3b4sd) [inline configuration/provider.go]: Done.

File: configuration/provider.go (hunk 2 above)
Comment (xh3b4sd) [inline configuration/provider.go]: Done.

**Pull request merged**